### PR TITLE
fix(general): Enable running cloned policies in case the OOTB policy is suppressed

### PR DESCRIPTION
**Pull request by tronxd**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

Handle cases where cloned policies are not ran in case their OOTB policy is suppressed

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/main.py [hunk 1]
```diff
@@ -326,8 +328,11 @@ def run(banner: str = checkov_banner, argv: list[str] = sys.argv[1:]) -> int | N
     logger.debug(f"Filtered list of policies: {runner_filter.filtered_policy_ids}")
 
     runner_filter.excluded_paths = runner_filter.excluded_paths + list(repo_config_integration.skip_paths)
-
-    runner_filter.set_suppressed_policies(suppressions_integration.get_policy_level_suppressions())
+    policy_level_suppression = suppressions_integration.get_policy_level_suppressions()
+    bc_cloned_checks = custom_policies_integration.bc_cloned_checks
```
Comment (bo156) [inline checkov/main.py:332]: you can remove this variable and save a line

File: checkov/main.py [hunk 2]
```diff
@@ -757,15 +762,20 @@ def run(self, banner: str = checkov_banner) -> int | None:
...
         logger.debug(f"Filtered list of policies: {runner_filter.filtered_policy_ids}")
 
         runner_filter.excluded_paths = runner_filter.excluded_paths + list(repo_config_integration.skip_paths)
-
-        runner_filter.set_suppressed_policies(suppressions_integration.get_policy_level_suppressions())
+        policy_level_suppression = suppressions_integration.get_policy_level_suppressions()
+        bc_cloned_checks = custom_policies_integration.bc_cloned_checks
+        runner_filter.bc_cloned_checks = bc_cloned_checks
+        custom_policies_integration.policy_level_suppression = policy_level_suppression
+        runner_filter.set_suppressed_policies(policy_level_suppression)
```
Comment (bo156) [inline checkov/main.py:778]: looks like you have this block of code 3 times here, can you extract to a function?

File: checkov/runner_filter.py [hunk 3]
```diff
@@ -207,15 +208,15 @@ def should_run_check(
...
+            (bc_check_id in self.suppressed_policies and bc_check_id not in self.bc_cloned_checks)
         )
         logging.debug(f'skip_severity = {skip_severity}, explicit_skip = {explicit_skip}, regex_match = {regex_match}, suppressed_policies: {self.suppressed_policies}')
         logging.debug(
             f'bc_check_id = {bc_check_id}, include_all_checkov_policies = {self.include_all_checkov_policies}, is_external = {is_external}, explicit_run: {explicit_run}')
 
         if should_skip_check:
             result = False
-            logging.debug(f'should_skip_check {check_id}: {result}')
+            logging.debug(f'should_skip_check {check_id}: {should_skip_check}')
```
Comment (bo156) [inline checkov/runner_filter.py:219]: we dont want the result?

**Review by bo156 — CHANGES_REQUESTED** [review]

File: checkov/runner_filter.py (hunk 3 above)
Comment (tronxd) [reply checkov/runner_filter.py:219]: It's the opposite of result

File: checkov/main.py (hunk 2 above)
Comment (tronxd) [reply checkov/main.py:778]: Not needed IMO (we won't reuse it anywhere else than main)

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 4]
```diff
@@ -215,10 +215,10 @@ def pre_runner(self, runner: _BaseRunner) -> None:
 
     def get_policy_level_suppressions(self) -> List[str]:
         policy_level_suppressions = []
-        for check_id, check_suppressions in self.suppressions.items():
+        for _, check_suppressions in self.suppressions.items():
```
Comment (gruebel) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:218]:
```suggestion
        for check_suppressions in self.suppressions.values():
```
🙂

**Review by bo156 — APPROVED** [review]

**Review by gruebel — APPROVED** [review]
🏖️

**Pull request merged**